openssl/src/crypto/pkey.rs

```diff
@@ -205,6 +205,30 @@ impl PKey {
         }
     }
 
+    /// pass ownership of the RSA key to this
+    pub fn set_rsa(&mut self, rsa: RSA) {
+        unsafe {
+            // TODO: should we do something like panic if null? this will fail silently right now
```

Comment:
based on your comment below about keeping the ffi::RSA pointer private in RSA, if that was hidden behind a ctor, as I suggested below, then I could make sure a null check is performed there which would reduce the chance of this being null to nil.

If I make that change, should I do a panicking null check?